cpg-library/src/main/java/de/fraunhofer/aisec/cpg/frontends/java/StatementAnalyzer.java

```diff
@@ -184,12 +184,11 @@ private ForEachStatement handleForEachStatement(Statement stmt) {
     de.fraunhofer.aisec.cpg.graph.statements.Statement iterable =
         lang.getExpressionHandler().handle(forEachStmt.getIterable());
 
-    if (!(var instanceof DeclarationStatement
-        && ((DeclarationStatement) var).isSingleDeclaration())) {
-      log.error("more or unknown decl in foreach");
-    } else {
-      statement.setVariable(((DeclarationStatement) var).getSingleDeclaration());
-    }
+    if (!(var instanceof DeclarationStatement))
+      log.error("Expected a DeclarationStatement but received: {}", var.getName());
+    else if (!((DeclarationStatement) var).isSingleDeclaration())
```

Comment:
I think you can remove this, I think it is ok to check whether this is a declaration statement, but any more checks are not necessary